### DISCUSSION TITLE
Read stdin from the goja?

### DISCUSSION THREAD

**Discussion by antonmedv** [dop251/goja#499] [Q&A] [answered]
I need to parse a big JSON file from the goja. File coming from stdin in Go. I don’t want to accumulate the file in Go to reduce memory footprint. 

Is is possible to write a sync function in goja to get a “chuck” of stdin? Read stdin from goja in a sync way?

Thanks.

**Comment by dop251** [comment, marked as answer]
Yes, you can `Set` any Go function on `*Runtime` or on an `*Object` and call it from javascript. Sync is the default mode of operations unless you use the EventLoop from goja_nodejs or similar.
